test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/integration/ProcessInstanceMigrationIntegrationTest.java

```diff
@@ -0,0 +1,229 @@
+        } catch (UnsupportedOperationException ex) {
+            logger.info("KIE Server + MySQL is skipped.", ex);
+        }
+
+        try {
+            KieServerWithDatabaseScenario kieServerPostgreSqlScenario = deploymentScenarioFactory.getKieServerWithPostgreSqlScenarioBuilder()
+                                                                                                 .withInternalMavenRepo(true)
+                                                                                                 .withContainerDeployment(KIE_CONTAINER_DEPLOYMENTS)
+                                                                                                 .withProcessMigrationDeployment()
+                                                                                                 .build();
+            KjarDeploymentScenarioListener.addKjarDeployment(kieServerPostgreSqlScenario, OLD_KJAR, NEW_KJAR);
+            scenarios.add(new Object[]{"KIE Server + PostgreSQL", kieServerPostgreSqlScenario});
+        } catch (UnsupportedOperationException ex) {
+            logger.info("KIE Server + PostgreSQL is skipped.", ex);
+        }
+
+        return scenarios;
+    }
+
+    @Override
+    protected KieServerWithDatabaseScenario createDeploymentScenario(DeploymentScenarioBuilderFactory deploymentScenarioFactory) {
+        return kieServerScenario;
+    }
+
+    @Before
+    public void setUp() throws KeyManagementException, NoSuchAlgorithmException, KeyStoreException {
+        logger.info("URL " + deploymentScenario.getKieServerDeployment().getUrl());
+        kieServicesClient = KieServerClientProvider.getKieServerClient(deploymentScenario.getKieServerDeployment());
+        processServicesClient = KieServerClientProvider.getProcessClient(deploymentScenario.getKieServerDeployment());
+
+        kieServerId = kieServicesClient.getServerInfo().getResult().getServerId();
+        client = HttpClientBuilder.create().setDefaultCredentialsProvider(getBasicAuth()).build();
+    }
+
+    @Test
+    public void testProcessInstanceMigrationIntegration() {
+        // Given
+        startProcesses();
+
+        // When
+        try {
+            createMigration();
+        } catch (IOException | JAXBException e) {
+            throw new RuntimeException("Process instance migration failed", e);
+        }
+
+        // Then
+        List<ProcessInstance> instances = processServicesClient.findProcessInstances(OLD_CONTAINER_ID, 0, 10);
+        Assertions.assertThat(instances).hasSize(1);
+        Assertions.assertThat(instances.get(0).getId()).isIn(instancesIds);//.isEqualTo(Long.valueOf(2L));
+
+        instances = processServicesClient.findProcessInstances(NEW_CONTAINER_ID, 0, 10);
+        Assertions.assertThat(instances).hasSize(1);
+        Assertions.assertThat(instances.get(0).getId()).isIn(instancesIds);//.isEqualTo(Long.valueOf(1L));
+    }
+
+    private Migration createMigration() throws IOException, JAXBException {
```

Comment:
Same comments as above. I think we can save a lot of code having a ProcessInstanceMigrationClient.